src/app/modules/app/pages/courses/components/course-form/course-form.component.ts

```diff
@@ -50,43 +41,47 @@ export class CourseFormComponent implements OnInit, OnDestroy {
     });
   }
 
-  onSubmit(): void {
+  public getFieldError(field: string): string | null {
+    const control = this.courseForm.get(field);
+    if (control?.hasError('required')) {
+      return 'This field is required.';
+    } else if (control?.hasError('maxlength')) {
+      const errors = control.errors ? control.errors['maxlength'] : null;
+      const maxLength = errors ? errors.requiredLength : 0;
+      return `This field cannot be longer than ${maxLength} characters.`;
+    }
+    return null;
+  }
+
+  public onSubmit(): void {
+    console.log(this.courseForm.valid);
     if (this.courseForm.valid) {
       if (this.data) {
-        this.editCourse(this.courseForm.value);
+        this.editCourse();
       } else {
-        this.createCourse(this.courseForm.value);
+        this.createCourse();
       }
-    }
+    } else this.courseForm.markAllAsTouched();
```

Comment:
missing { after else ?

Reply:
No, an if statement (and else afterwards) execute the inmediate next line of code if not followed by curly braces ({})
example:
if(true)
  doSomething();
else
  doSomethingElse()

Reply:
you're rigth. I was confused by the } at line 65